test/common/testlib.py

```diff
@@ -476,16 +476,25 @@ def logout(self):
         if self.is_visible("button#machine-reconnect"):
             # happens when shutting down cockpit or rebooting machine
             self.click("button#machine-reconnect")
+            self.expect_load()
+            return False
         else:
             # happens when cockpit is still running
             self.click("#navbar-dropdown")
             self.click('#go-logout')
-        self.expect_load()
+            self.expect_load()
+            self.expect_load()
+        return True
 
     def relogin(self, path=None, user=None, authorized=None):
         if user is None:
             user = self.default_user
-        self.logout()
+        if self.logout():
+            href = path
+            if not href:
+                href = "/"
+            self.open(href)
```

Comment:
This at least requires a comment. Why does it have to explicitly open the given path now?